build fails on Frontier with undefined reference to MPL_gpu_ipc_handle_destroy

I've adapted the instructions for building on Crusher (https://github.com/pmodels/mpich/wiki/Using-MPICH-on-Crusher@OLCF) in order to try to build MPICH on Frontier.
```
git clone --recursive git@github.com:pmodels/mpich.git
cd mpich
module load rocm/5.1.0
module load libtool
./autogen.sh
./configure --with-device=ch4:ofi --with-libfabric=/opt/cray/libfabric/1.15.2.0 --with-hip=$ROCM_PATH/hip
make -j8
```

However, it fails at the link stage:
```
  GEN      lib/libmpi.la
copying selected object files to avoid basename conflicts...
  GEN      lib/libmpifort.la
  CXXLD    lib/libmpicxx.la
  CCLD     src/env/mpichversion
  CCLD     src/env/mpivars
//usrusr//lib64lib64//gccgcc//x86_64x86_64--susesuse--linuxlinux//77//....//....//....//....//x86_64x86_64--susesuse--linuxlinux//binbin//ldld::  liblib//..libslibs//libmpi.solibmpi.so::  undefinedundefined  referencereference  toto  ``MPL_gpu_ipc_handle_destroyMPL_gpu_ipc_handle_destroy''

collect2: error: ld returned 1 exit status
collect2: error: ld returned 1 exit status
```

--- a/src/mpl/src/gpu/mpl_gpu_hip.c
+++ b/src/mpl/src/gpu/mpl_gpu_hip.c
@@ -132,6 +132,11 @@ int MPL_gpu_ipc_handle_create(const void *ptr, MPL_gpu_ipc_mem_handle_t * ipc_ha
     goto fn_exit;
 }
 
+int MPL_gpu_ipc_handle_destroy(const void *ptr)
+{
+    return MPL_SUCCESS;
+}
+
 int MPL_gpu_ipc_handle_map(MPL_gpu_ipc_mem_handle_t ipc_handle, int dev_id, void **ptr)
 {
     int mpl_err = MPL_SUCCESS;

